pkg/chunk/chunk_store_utils.go

```diff
@@ -230,22 +242,32 @@ func (c *Fetcher) processCacheResponse(ctx context.Context, chunks []Chunk, keys
 
 	go func() {
 		for _, request := range requests {
-			c.decodeRequests <- request
+			select {
+			case <-c.quit:
+				return
+			default:
```

Comment:
Can this happen: `quit` is not closed, so we go to `default` and block on send.
Then `quit` is closed and all workers exit.
Now we are hung.
Should the chan send be brought up to the `select`, so either can proceed each time round the loop?

Reply:
I was a bit unsure about if we could use both send and receive cases in the same select, but seems to be OK since the channels are read/closed from other goroutines, so I moved it up to a case.